### PR TITLE
feat: manual instrumentation using Python

### PR DESCRIPTION
**Pull request by pmm-sumo**
Manual instrumentation guide

### CODE REVIEW

File: exercises/instrumentation/logs/Dockerfile-server [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+COPY . .
+RUN pip install -r requirements.txt
```
Comment (perk-sumo) [inline exercises/instrumentation/logs/Dockerfile-server:7]:
```suggestion
COPY requirements.txt .
RUN pip install -r requirements.txt

COPY . .
```

File: exercises/instrumentation/logs/Dockerfile-client [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+COPY . .
+RUN pip install -r requirements.txt
```
Comment (perk-sumo) [inline exercises/instrumentation/logs/Dockerfile-client:7]:
```suggestion
COPY requirements.txt .
RUN pip install -r requirements.txt

COPY . .
```

File: exercises/instrumentation/metrics/Dockerfile-client [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+COPY . .
+RUN pip install -r requirements.txt
```
Comment (perk-sumo) [inline exercises/instrumentation/metrics/Dockerfile-client:7]:
```suggestion
COPY requirements.txt .
RUN pip install -r requirements.txt

COPY . .
```

File: exercises/instrumentation/metrics/Dockerfile-server [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+COPY . .
+RUN pip install -r requirements.txt
```
Comment (perk-sumo) [inline exercises/instrumentation/metrics/Dockerfile-server:7]:
```suggestion
COPY requirements.txt .
RUN pip install -r requirements.txt

COPY . .
```

File: exercises/instrumentation/tracing/Dockerfile-client [hunk 5]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+COPY . .
+RUN pip install -r requirements.txt
```
Comment (perk-sumo) [inline exercises/instrumentation/tracing/Dockerfile-client:7]:
```suggestion
COPY requirements.txt .
RUN pip install -r requirements.txt

COPY . .
```

File: exercises/instrumentation/tracing/Dockerfile-server [hunk 6]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+COPY . .
+RUN pip install -r requirements.txt
```
Comment (perk-sumo) [inline exercises/instrumentation/tracing/Dockerfile-server:7]:
```suggestion
COPY requirements.txt .
RUN pip install -r requirements.txt

COPY . .
```

File: exercises/instrumentation/logs/README.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    logging.getLogger().addHandler(handler)
+```
+
+### Actual initialization
+
+With the above implemented, it's not just a matter of importing the function and calling it, e.g.
+
+```python
+setup_log_provider()
+```
```
Comment (perk-sumo) [inline exercises/instrumentation/logs/README.md:103]: Maybe we could add info on how to do the actual import as well?

```python
from common.ot import setup_log_provider
```

File: exercises/instrumentation/logs/README.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    # Attach it
+    logging.getLogger().addHandler(handler)
+```
+
+### Actual initialization
+
+With the above implemented, it's not just a matter of importing the function and calling it, e.g.
+
+```python
+from ot import setup_log_provider
```
Comment (perk) [inline exercises/instrumentation/logs/README.md:97]:
```suggestion
from common.ot import setup_log_provider
```

Above you mention `common/ot.py` :)

File: exercises/instrumentation/tracing/Dockerfile-client [hunk 9]
```diff
@@ -0,0 +1,9 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+RUN pip install -r requirements.txt
```
Comment (perk) [inline exercises/instrumentation/tracing/Dockerfile-client:6]:
```suggestion
COPY requirements.txt requirements.txt
RUN pip install -r requirements.txt
```

File: exercises/instrumentation/tracing/Dockerfile-server [hunk 10]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.9-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash
+
+RUN pip install -r requirements.txt
```
Comment (perk) [inline exercises/instrumentation/tracing/Dockerfile-server:6]:
```suggestion
COPY requirements.txt requirements.txt
RUN pip install -r requirements.txt
```

File: exercises/instrumentation/logs/README.md (hunk 8 above)
Comment (pmm-sumo) [reply exercises/instrumentation/logs/README.md:97]: Yeah, I got rid of it, let me fix the mention

File: exercises/instrumentation/tracing/Dockerfile-client (hunk 9 above)
Comment (perk) [reply exercises/instrumentation/tracing/Dockerfile-client:6]: Hm, was that on purpose? 🤔 I'm not sure anymore 😅

Comment (pmm-sumo) [reply exercises/instrumentation/tracing/Dockerfile-client:8]: nah, my bad, fixed few other places as well

**Review by perk — APPROVED** [review]
LGMT 🏅

**Pull request merged**